src/xenia/kernel/kernel_state.cc

```diff
@@ -575,5 +592,60 @@ void KernelState::CompleteOverlappedDeferredEx(
   dispatch_cond_.notify_all();
 }
 
+bool KernelState::Save(ByteStream* stream) {
+  XELOGD("Serializing the kernel...");
+  stream->Write('KRNL');
+
+  // Save the object table
+  object_table_.Save(stream);
+
+  // Save all objects
+  auto objects = object_table_.GetAllObjects();
+  uint32_t* num_objects_ptr = (uint32_t*)(stream->data() + stream->offset());
+  size_t num_objects = objects.size();
+  stream->Write((uint32_t)num_objects);
```

Comment:
nit: prefer static_cast and reinterpret_cast and such instead of C-style casts. At some point I'd like to turn on cpplint to warn on this stuff, so just a heads up :)